challengeA/Amagata/Simulator.rb

```diff
@@ -0,0 +1,131 @@
+# 計算するクラス
+class Calculator
+  def initialize(ampare,kilowatt_hour)
+    @ampare=ampare
+    @kilowatt_hour=kilowatt_hour
+  end
+
+  def calculate
+    # Informationクラスをインスタンス化
+    info=Information.new
+    # informationクラスのゲッターを配列に入れる。（手動）
+    arrays_all=[info.tepco,info.loooop,info.tokyo_gas]  # 会社を追加した場合ここに新料金を足さなければならない。,info.amagata
+    price=[]
+    # 条件から各社料金を計算
+    arrays_all.each do |array|
+      if array[:contract_current].include?(@ampare)
+        # 基本料金を定義 アンペアがbasic_chargeにあればその基本料金料金、なければ0
+        if info.basic_charge[:"#{@ampare}A"]
+          basic = info.basic_charge[:"#{@ampare}A"]
+        else
+          basic = 0
+        end
+        # 単価決定の条件式1 
+        if array[:range][0]<= @kilowatt_hour && @kilowatt_hour < array[:range][1]
+          # 料金=基本料金+従量課金
+          sum = basic + array[:pay_per_use][0] * @kilowatt_hour
+        # 単価決定の条件式2
+        elsif @kilowatt_hour >= array[:range][1] && @kilowatt_hour < array[:range][2]
+          sum = basic + array[:pay_per_use][1] * @kilowatt_hour
+        # 単価決定の条件式3
+        else
+          sum = basic + array[:pay_per_use][2] * @kilowatt_hour
+        end
+      else
+        sum = 0
+      end
+      price << sum
+    end
+
+    # 料金の計算結果の個数だけ会社とプラント料金を、配列@resultsに入れる。
+    @results=[]
+    price.length.times do |i|
+      if price[i] != 0
+        @results << { provider_name: "#{arrays_all[i][:provider_name]}", plan_name: "#{arrays_all[i][:plan_name]}", price: "#{price[i]}" }
+      end
+    end
```

Comment:
pricesにはアンペアに対応するプランのみを保持する様にしておき、
```suggestion
    prices.map.with_index do |price, i|
      { provider_name: "#{arrays_all[i][:provider_name]}", plan_name: "#{arrays_all[i][:plan_name]}", price: price }
    end
```
あたりでどうでしょう・・・？（動作未確認）

Reply:
このヒントをもとに、このコードの前で計算をしている配列arrays_allに.map.with_indexを適応することで41-43行目コードを無くすことができました。

ありがとうございます